File: tools/build_static.mjs
import { cp, mkdir, rm } from "node:fs/promises";
import path from "node:path";
import { fileURLToPath } from "node:url";

const root = path.resolve(path.dirname(fileURLToPath(import.meta.url)), "..");
const dist = path.join(root, "dist");

async function copyFile(source, destination = source) {
  const target = path.join(dist, destination);
  await mkdir(path.dirname(target), { recursive: true });
  await cp(path.join(root, source), target);
}

await rm(dist, { force: true, recursive: true });
await mkdir(dist);

for (const file of ["index.html", "styles.css", "app.js", "_routes.json"]) {
  await copyFile(file);
}

await cp(path.join(root, "data"), path.join(dist, "data"), { recursive: true });
console.log(`Built ${dist}`);

File: tools/build_static.py
#!/usr/bin/env python3
"""Create the Cloudflare Pages static output directory."""

from __future__ import annotations

import shutil
from pathlib import Path


ROOT = Path(__file__).resolve().parents[1]
DIST = ROOT / "dist"


def copy_file(source: str, destination: str | None = None) -> None:
    destination_path = DIST / (destination or source)
    destination_path.parent.mkdir(parents=True, exist_ok=True)
    shutil.copy2(ROOT / source, destination_path)


def main() -> None:
    if DIST.exists():
        shutil.rmtree(DIST)
    DIST.mkdir()

    for file_name in ("index.html", "styles.css", "app.js", "_routes.json"):
        copy_file(file_name)

    shutil.copytree(ROOT / "data", DIST / "data")
    print(f"Built {DIST}")


if __name__ == "__main__":
    main()
